src/components/RetipComponent.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+<div class="position-relative wrapper">
+  <img @click="toggleRetip(!show)" class="retip__icon" src="../assets/heart.svg">
+  <div class="clearfix retip__container" v-if="show">
+    <div class="input-group mr-1 float-left">
+      <input type="number" step="0.000001" v-model="value" @input="onValueChange" class="form-control" aria-label="Default">
+      <div class="input-group-append">
+        <span class="input-group-text append__ae"> <span class="ae">AE</span>&nbsp;<fiat-value :amount="value"></fiat-value></span>
+      </div>
+    </div>
+    <button class="btn btn-primary retip__button float-right" @click="retip()">Retip</button>
+  </div>
+</div>
+
+</template>
+
+<script>
+  import { mapGetters } from 'vuex'
+  import util from '../utils/util';
+  import Currency from '../utils/currency'
+  import BigNumber from 'bignumber.js'
+  import aeternity from '../utils/aeternity';
+  import {EventBus} from '../utils/eventBus';
+  import FiatValue from './FiatValueComponent.vue'
+
+  export default {
+    name: 'RetipComponent',
+    props: ['tipid'],
+    data() {
+      return {
+        fiatValue: 0.00,
+        value: 0,
+        show: false
+      }
+    },
+    components: {
+      'fiat-value': FiatValue
+    },
+    computed: {
+      ...mapGetters(['current']),
+    },
+    methods:{
+      toggleRetip(flag){
+        this.show = flag;
+        if(flag){
+          this.resetForm();
+        }
+      },
+      async retip() {
+        let amount = util.aeToAtoms(this.value);
+        await aeternity.contract.methods.retip(this.tipid, {amount: amount}).catch(console.error);
+        EventBus.$emit('reloadData');
+        this.show = false;
+      },
+      resetForm(){
+        this.value = 0;
+        this.fiatValue = 0.00;
+      },
+      onValueChange(data){
+        this.getFiatVal();
+      },
+      async getFiatVal(){
```

Comment:
whats the need for all the fiat val logic in here? there is a fiat val component.